npm/private/yaml.bzl

```diff
@@ -81,6 +81,36 @@ def _handle_CONSUME_SPACE(state, input, stack, starlark):
         # Consume any space following the {
         stack.append(_new_CONSUME_SPACE_FLOW())
 
+    elif input == "?":
```

Comment:
So these key/value pairs are initiated different then the standard ":" style. With the ":" we might not know we are in a map until we reach the ":", and at that point we've already parsed the key. While the "?" comes before the key so we know up front we're in a map, but we  don't know the key yet.

Reply:
nit: add a comment here that the parser currently only supports string keys for complex maps